### ISSUE TITLE
App crash: NullPointerException

### ISSUE THREAD

**Issue by OlehHaidaienko** [firebase/firebase-android-sdk#4146]
### [REQUIRED] Step 2: Describe your environment

  * Android Studio version: Android Studio Dolphin | 2021.3.1
  * Firebase Component: Performance
  * Component version: 20.1.1

#### Steps to reproduce:

After updating Firebase BoM from `29.2.0`  to `30.4.1` we started receiving a lot of crashes on Android 7, 8, and 9. Unfortunately, we can't reproduce this issue. 

Here is the crash log
```
Caused by java.lang.NullPointerException: Attempt to invoke virtual method 'int java.util.ArrayList.size()' on a null object reference
       at android.view.View.findFrameMetricsObserver(View.java:6539)
       at android.view.View.removeFrameMetricsListener(View.java:6510)
       at android.view.Window.removeOnFrameMetricsAvailableListener(Window.java:886)
       at androidx.core.app.FrameMetricsAggregator$FrameMetricsApi24Impl.remove(FrameMetricsAggregator.java:433)
       at androidx.core.app.FrameMetricsAggregator.remove(FrameMetricsAggregator.java:243)
       at com.google.firebase.perf.application.FrameMetricsRecorder.stop(FrameMetricsRecorder.java:110)
       at com.google.firebase.perf.application.AppStateMonitor.sendScreenTrace(AppStateMonitor.java:355)
       at com.google.firebase.perf.application.AppStateMonitor.onActivityStopped(AppStateMonitor.java:210)
       at android.app.Application.dispatchActivityStopped(Application.java:257)
       at android.app.Activity.onStop(Activity.java:1914)
       at androidx.fragment.app.FragmentActivity.onStop(FragmentActivity.java:496)
       at androidx.appcompat.app.AppCompatActivity.onStop(AppCompatActivity.java:252)
       at com.fusionmedia.investing.ui.activities.base.BaseSplashActivity.onStop(BaseSplashActivity.java:922)
       at android.app.Instrumentation.callActivityOnStop(Instrumentation.java:1433)
       at android.app.Activity.performStop(Activity.java:7367)
       at android.app.ActivityThread.callActivityOnStop(ActivityThread.java:4146)
       at android.app.ActivityThread.performStopActivityInner(ActivityThread.java:4124)
       at android.app.ActivityThread.handleStopActivity(ActivityThread.java:4199)
       at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:192)
       at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:165)
       at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:142)
       at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:70)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1811)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loop(Looper.java:193)
       at android.app.ActivityThread.main(ActivityThread.java:6694)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
```
**BaseSplashActivity.java:922** looks like this:
```
@Override 
protected void onStop() {
    super.onStop() → Crash happens here
    ....
}
```

<img width="1273" alt="image" src="https://user-images.githubusercontent.com/7819367/192446033-83053383-387b-46cb-990b-56ec094c0df6.png">

After rollback to Firebase BoM `29.2.0` issue is gone

**Comment by argzdev** [comment]
Thanks for reporting, @OlehHaidaienko. I'll let our engineers know about this.

Just some notes, crash happened here:
https://github.com/firebase/firebase-android-sdk/blob/1d344e8f407f9f6ee7d8510581d516a286f29cbc/firebase-perf/src/main/java/com/google/firebase/perf/application/FrameMetricsRecorder.java#L97

https://github.com/firebase/firebase-android-sdk/blob/1d344e8f407f9f6ee7d8510581d516a286f29cbc/firebase-perf/src/main/java/com/google/firebase/perf/application/AppStateMonitor.java#L348

**Comment by jeremyjiang-dev** [comment]
Could be related to b/133827763

**Comment by kimrtaylor** [comment]
Just wanted to add that we are also experiencing this crash.

It looks like there are some PRs in progress

 @argzdev  Do we know if there will be a fix available soon? That will help us decide whether to rollback, or hold off for the  fix. Thanks a lot 🙏

**Comment by argzdev** [comment]
Hi @kimrtaylor, upon checking, it looks like the PR is still in progress, and it may take some time before it is merged in the main branch. That said, once the fix is merged we would still expect this fix to be in the M124 release that is targeted for the week of Nov. 17 (around the third week). 

Since there's still a long wait, I'd advise to rollback for now. Thanks!

**Comment by kimrtaylor** [comment]
Great that's exactly what we needed to know, thanks @argzdev , we will rollback while we wait for the fix

**Comment by leotianlizhan** [comment]
Should be fixed in https://github.com/firebase/firebase-android-sdk/pull/4284, which is released in Firebase Performance 20.3.0, part of Firebase BOM 31.1.0.

**Comment by argzdev** [comment]
Thanks for the update, @leotianlizhan. I'll close this now.
